### PR TITLE
Fuzzers for ed25519 signature verification

### PR DESCRIPTION
**Pull request by theo25**
This PR adds 3 fuzzers for the ed25519 signing and signature verification APIs of Firedancer.
- `fuzz_ed25519_sign`: A fuzzer for the input of the signing API (fuzzes the private key and message)
- `fuzz_ed25519_verify)`: A fuzzer for the input of the signature verification API (fuzzes the signature, public key, and message)
- `fuzz_ed25519_sigverify`: A fuzzer that fuzzes the input to the signing API and then checks the generated signature with the verification API (fuzzes the private key and message)

The `fuzz_ed25519_sigverify` fuzzer generated an input that failed verification in my local testing. Following is the crash file in text form:
```
$ xxd crash-9857cdc91fc9079567d4d02b5fcda31c8d7a67be                                   
00000000: 0a4d 4d4d adb6 4d4d 4d00 4d4d 4d4d 4d4d  .MMM..MMM.MMMMMM
00000010: 4d2a 2a2a 2a2a 2a4d 4d4d 4d4d 4d4d 4dfe  M******MMMMMMMM.
00000020: ffff ff4d 4d4d 4d4d 4d4d                 ...MMMMMMM
```

### CODE REVIEW

File: src/ballet/ed25519/fuzz_ed25519_sigverify.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  fd_sha512_t _sha[1];
+  fd_sha512_t *sha = fd_sha512_join( fd_sha512_new( _sha ) );
+
+  uchar pub[ 32 ];
+  fd_ed25519_public_from_private( pub, test->prv, sha );
+
+  uchar sig[ 64 ];
+  void * sig_result = fd_ed25519_sign( sig, test->msg, sz, pub, test->prv, sha );
+  int cmp = strncmp( ( char * ) sig, ( char * ) sig_result, 64UL );
```
[the remark this answers is not in the document]
Comment (lheeger-jump) [reply src/ballet/ed25519/fuzz_ed25519_sigverify.c:43]: mmm yes if there is 0 byte in the signature this would be wrong

**Comment by marcus-ares** [comment]
Please add the crash file to `corpus/fuzz_ed25519_sigverify/` :)

**Comment by theo25** [comment]
I have added the crash file as requested. I will also fix the issue with `strncmp` soon, thanks for pointing that out!

File: src/ballet/ed25519/fuzz_ed25519_sigverify.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (theo25) [reply src/ballet/ed25519/fuzz_ed25519_sigverify.c]: I just pushed a fix for this.

**Review by marcus-jump — APPROVED** [review]

**Pull request merged**